media/server/main/source/MediaKeySession.cpp

```diff
@@ -121,28 +122,18 @@ MediaKeyErrorStatus MediaKeySession::generateRequest(InitDataType initDataType,
         }
         return status;
     }
-    else
-    {
-        // Get the challenge manually and notify onLicenseRequest
-        getChallenge();
-    }
 
     return MediaKeyErrorStatus::OK;
 }
 
 void MediaKeySession::getChallenge()
 {
+    RIALTO_SERVER_LOG_DEBUG("entry:");
     auto task = [&]()
     {
         uint32_t challengeSize = 0;
         MediaKeyErrorStatus status = m_ocdmSession->getChallengeData(m_kIsLDL, nullptr, &challengeSize);

```

Comment:
Why are we ignoring the return status?

Reply:
That's one of differences between OCDM for PR 2.5 and PR 4.0 :). As you can see, getChallengeData has to be called twice. First call is to find out what is the size of the challenge, second call is to get the challenge. First call for PR 4.0 OCDM always fails with ERROR_OUT_OF_MEMORY, it is also skipped in "real" netflix